rust/connlib/snownet/src/node.rs

```diff
@@ -108,6 +108,10 @@ where
         }
     }
 
+    pub fn public_key(&self) -> PublicKey {
+        (&self.private_key).into()
```

Comment:
I kind of understand why boringtun would make this API choice, but at a glance it looks like it's transmuting a private key to a public key, that threw me off for a couple minutes until I found the docs.

https://docs.rs/boringtun/latest/boringtun/x25519/struct.PublicKey.html#impl-From%3C%26'a+EphemeralSecret%3E-for-PublicKey

Reply:
TBH, I am not a fan of these `From` impls. I'd rather have explicit functions like `to_public_key`, esp. because those can easily be defined on `&self` and then you don't need to do this trickery of using parenthesis before calling `into`.